apiserver/facade/registry.go

```diff
@@ -30,9 +28,8 @@ type Registry struct {
 }
 
 // RegisterStandard is the more convenient way of registering
-// facades. newFunc should have one of the following signatures:
+// facades. newFunc should have one of the following signature:
 //   func (facade.Context) (*Type, error)
-//   func (*state.State, facade.Resources, facade.Authorizer) (*Type, error)
 func (f *Registry) RegisterStandard(name string, version int, newFunc interface{}) error {
```

Comment:
I believe we chould change `newFunc interface{}` to be `newFunc func(Context) (interface{}, error)` and all the reflect code can be removed. That way we can follow use compiler checking instead of runtime checking.